Add tiny maccalc helper tool

mhei opened the pull request:
This adds a tiny helper tool for handling MAC addresses in user-space.

It can be helpful during manufacturing process and in scripts when updating e.g. PIBs or similar stuff.

mhei committed:
packagegroup-common: add maccalc helper tool

Signed-off-by: Michael Heimpold <mhei@heimpold.de>

barsnick commented on recipes-devtools/maccalc/maccalc.bb:
```diff
@@ -0,0 +1,21 @@
+# This file was derived from the 'Hello World!' example recipe in the
+# Yocto Project Development Manual.
+#
+SUMMARY = "MAC address manupulation utility"
```
Typo: manipulation. OTOH. it's direct from the author's mouth.
```suggestion
SUMMARY = "MAC address manipulation utility"
```

mhei replied:
C&P without noticing... will fix anyway

barsnick commented on recipes-devtools/maccalc/maccalc.bb:
```diff
@@ -0,0 +1,21 @@
+
+do_install() {
+    install -d ${D}${sbindir}
+    install -m 0755 maccalc ${D}${sbindir}
```
This tool doesn't actually manipulate the system, but rather does calculations. So it's fully appropriate for the regular user, and therefore doesn't feel correctly placed in sbindir.

lategoodbye commented on recipes-devtools/maccalc/maccalc/main.c:
```diff
@@ -0,0 +1,256 @@
+	i = atoi(argv[1]);
+
+	t = (mac[3] << 16) | (mac[4] << 8) | mac[5];
+	t += i;
```
Shouldn't we care about overflow?

barsnick replied:
The code probably assumes that an overflow into the OUI is not a use case.

mhei replied:
I agree, that the upstream code has room for improvement, but for now, I'd re-use it as-is. And I agree that an overflow into the OUI should not happen.

lategoodbye commented:
General question: it was intended that you contribute via your private mail address?

barsnick reviewed (changes requested):
I didn't review the actual tool, assuming it's safe and works fine, and just begs to be included.

barsnick commented on recipes-devtools/maccalc/maccalc.bb:
```diff
@@ -0,0 +1,21 @@
+S = "${WORKDIR}"
+
+do_compile() {
+    ${CC} ${LDFLAGS} main.c -o maccalc
```
I'm not sure whether bitbake defines generic CFLAGS which we should use, and whether it places them in `$CFLAGS` or embeds them in `$CC`.

mhei commented:
> General question: it was intended that you contribute via your private mail address?

Preparation and initial testing was done outside of working hours. Will change the SoB in edited commits.

barsnick commented on recipes-devtools/maccalc/maccalc.bb:
```diff
@@ -0,0 +1,21 @@
+SUMMARY = "MAC address manipulation utility"
+
+LICENSE = "GPL-2.0-only"
+LIC_FILES_CHKSUM = "file://main.c;md5=641b6faf83ff4caf4264dde86a5ef503"
```
I just noticed some more room for nitpicking:
What I've seen in Yocto is that the license checksum is calculated over a few lines of a file, to just catch possible changes of the license (not in the other sections of text) when changing the sources:
```suggestion
LIC_FILES_CHKSUM = "file://main.c;beginline=6;endline=8;md5=badf00d"
```
See e.g. https://git.openembedded.org/meta-openembedded/tree/meta-oe/recipes-devtools/dmalloc/dmalloc_5.5.2.bb?h=master#n16

barsnick reviewed (approved):
Nit:
There's now a mismatch on the one commit between the commit's author email and the SOB footer. This would fail the standard GitHub CSO check (been there, had that). If this PR wasn't being squashed anyway, you would need to change the author of the commit which you changed the SOB on (e.g. with `git --amend --author="MH <...@chargebyte.com>"`).

I personally don't care, and find it appropriate to use your respective corresponding accounts.

mhei committed:
maccalc: add tiny helper tool for MAC address handling

This tool is handy for some usual MAC address handling tasks
in user space.

The source code is imported from
https://github.com/openwrt/packages/blob/7a07dfc6db5952e11e09bd9e2235774cd237fc15/net/maccalc/src/main.c

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>
